fixtures/enum-types/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+enum Animal {
+    Dog,
+    Cat,
+}
+
+// Though it has the proc-macro, we drop the variant
+// literals if there is not a repr type defined
+#[derive(uniffi::Enum)]
+pub enum AnimalInt {
+    Dog = 3,
+    Cat = 4,
+}
+
+#[repr(u8)]
+#[derive(uniffi::Enum)]
+pub enum AnimalUInt {
+    Dog = 3,
+    Cat = 4,
+}
+
+
+#[repr(u64)]
+#[derive(uniffi::Enum)]
+pub enum AnimalLargeUInt {
+    Dog = 123456789,
+    Cat = 987654321,
+}
+
+
+// TODO: Everything is treated as unsigned right now so
```

Comment:
I think we should uncomment this as otherwise we risk a breaking change - IIUC, this enum probably *does* work today, but without the correct discr being used by bindings. IIUC, this patch will mean the enum doesn't work at for kotlin?

Similarly, maybe add an enum with `#[repr(C, u8)]` (actually, while [the docs](https://doc.rust-lang.org/reference/type-layout.html#reprc-enums-with-fields) say it works, it now generates a warning, so I guess ignore this :)